### ISSUE TITLE
The size of _cython/cygrpc.so becomes more than x10 times larger in v1.27.1 compared to v1.26.0 and unable to deploy to GAE

### ISSUE THREAD

**Issue by kimamula** [grpc/grpc#21958]
### What version of gRPC and what language are you using?

Version of gRPC: v1.27.1
Language: Python

### What did you see?

Deployment to GAE results in the following error:

```sh
ERROR: (gcloud.app.deploy)
Cannot upload file [/path/to/lib/grpc/_cython/cygrpc.so], which has size [61958112] (greater than maximum allowed size of [33554432]).
Please delete the file or add to the skip_files entry in your application .yaml file and try again.
```

When I was using v1.26.0, the size of the file was ~6,000,000 and this error did not occur.

**Comment by kimamula** [comment]
Well, I am sorry that probably "x10 times larger" is not correct.
It seems that the size of cygrpc.so significantly differs depending on OS and I confirmed the size for v1.26.0 on my macOS and for v1.27.1 on the GAE log.

Moreover, specifying the file in `skip_files` as described in the log worked for me.
Therefore I close this issue.

**Comment by gnossen** [comment]
@kimamula What python interpreter version are you using?

**Comment by kimamula** [comment]
@gnossen I am using GAE standard environment for Python 2.7.

**Comment by jkevingutierrez** [comment]
Hi, 

I'm having the same problem:

ERROR: (gcloud.app.deploy) Cannot upload file [/out/grpc/_cython/cygrpc.so], which has size [52766863] (greater than maximum allowed size of [33554432]). Please delete the file or add to the skip_files entry in your application .yaml file and try again.

Deleting, or skipping the file works, but I was wondering if the file isn't needed? If that's the case, why is it generated? If it is needed, is there any way to reduce the size of that file?

**Comment by lidizheng** [comment]
@kimamula @jkevingutierrez Is it possible to upgrade to newer Python version? `3.6` and `3.7` should have the striped binaries.

We are working on a new release to fix this issue.

**Comment by bastostiago** [comment]
Hi all,

We have exactly the same error as mentioned. Builds via macOS or windows are working properly, but on linux machines (which is the most common CI infrastructure) the file becomes more than x10 times larger.

That's impossible to migrate to Python3 at the moment, so we are kind of desperate looking for a solution.

**Comment by lidizheng** [comment]
@bastostiago The release is still building, sorry for introducing this regression. There is a hacky way that might work, you can try to `strip` it manually:

```bash
strip --strip-debug cygrpc.so
```

Python packages are zip files. They can be unzip, modified, and then put back together (technically, you also need to update checksum).

**Comment by kimamula** [comment]
My colleague pointed out that the following lines in [GAE SE for Python 2 doc](https://cloud.google.com/appengine/docs/standard/python/runtime#Pure) likely indicate we can safely skip *.so files from deployment to GAE.

> All code for the Python runtime environment must be pure Python, and not include any C extensions or other code that must be compiled.

> Also ensure that your libraries do not need any [C extensions](https://docs.python.org/2/extending/extending.html)

Is that correct?

I have another question. I found documents ([here](https://cloud.google.com/appengine/docs/standard/python/tools/built-in-libraries-27) and [here](https://cloud.google.com/appengine/docs/standard/python3/migrating-to-cloud-ndb#installing_the_cloud_ndb_library_for_python_2_apps)) indicating that dependency to grpcio should be listed in app.yaml and the version should be fixed to 1.0.0 or 1.1.0dev0.
Maybe that is the right way to avoid this issue?

> @kimamula @jkevingutierrez Is it possible to upgrade to newer Python version? 3.6 and 3.7 should have the striped binaries.

Though we are working on upgrading Python, it will take several months more 😅

**Comment by nicolasnoble** [comment]
>That's impossible to migrate to Python3 at the moment, so we are kind of desperate looking for a solution.

> Though we are working on upgrading Python, it will take several months more 😅

I'm sorry if I am going to sound a bit harsh here, but Python 2 is officially deprecated since a few weeks now, and we've announced about a year ago that we would stop supporting it. There was ample time for anyone to migrate, and we won't have python 2 packages being produced much longer.

**Comment by lidizheng** [comment]
This issue is fixed since v1.27.2.
